### PR TITLE
Pytorch mnist refactor

### LINKED ISSUE #1214: Undefined variable in pytorch_mnist.ipynb
https://github.com/awslabs/amazon-sagemaker-examples/blob/7262f9eca92a612423dba16595161c2a473b771e/sagemaker-python-sdk/pytorch_mnist/pytorch_mnist.ipynb#L232

The variable `data` was never defined and thus the last cell before the cleanup section cannot run.

### PR DESCRIPTION
**Pull request by hsl89**
*Issue #, if available:*
fix #1214 

*Description of changes:*

1.Split training and inference into different notebooks
2.entry points train.py and inference.py reside in code/
3.use samples from test set to validate the endpoints
4.when testing the endpoint, we download mnist directly from Yann LeCun's website and convert it to numpy array of appropriate shapes depending on the deep learning framework

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by hsl89** [comment]
#1704 includes all changes in this PR
